corehq/apps/app_manager/forms.py

```diff
@@ -93,8 +96,22 @@ def clean(self):
             if link and link[0].master_domain != self.from_domain:
                 raise forms.ValidationError(
                     "The target project space is already linked to a different domain")
+
+        if self.cleaned_data['toggles'] and settings.SERVER_ENVIRONMENT != 'staging':
+            self._check_edit_permission_for_toggles()
+
         return self.cleaned_data
 
+    def _check_edit_permission_for_toggles(self):
+        toggle_slugs = set(self.cleaned_data['toggles'].split(','))
+        editable_toggle_slugs = {
```

Comment:
isn't there a function that gives you this already?

Reply:
okay this is toggle slug and not tag slug, phew, this could get confusing down the road.